docs/DeveloperGuide.md

```diff
@@ -1,38 +1,662 @@
+</code>
+
+- `[date]` is in `DD-MM-YYYY` format representing the date of the appointment.
+- `[time]` is in `HH:mm` format representing the time of the appointment.
+- `[description]` is a string  representing the details of the appointment. The string can only contain alphanumeric characters and spaces.
+
+##### Appointment Sequence
+
+1. User input is passed to `Handler.processInput()`, which determines the command used is `health`, thus passing the input to `Handler.handleHealth()`.
+
+2. `Handler.handleHealth()` determines the type of health which is `appointment`, and calls the `Parser.parseAppointmentInput()` method to process the user's input.
+
+3. `Parser.parseAppointmentInput()` splits the input using `Parser.splitAppointmentDetails()`. It then validates each input using `Validation.validateAppointmentDetails()`.
+    - `CustomExceptions.InsufficientInput` is thrown if either not enough parameters are specified or blank parameters are found.
+    - `CustomExceptions.InvalidInput` is thrown if the parameters passed in are invalid and do not follow the stipulated format.
+
+4. If valid, a new `Appointment` object is created with the split user input.
+
+5. The `Appointment` constructor adds the newly created object into `HealthList.APPOINTMENTS`.
+
+6. The `Appointment` object is passed to `Output.printAddAppointment()` and a message acknowledging the successful adding is printed to the screen.
+
+![AddAppointment](img/appointment_sequence.png)
+
+###### [Back to table of contents](#table-of-contents)
+
+#### Make Period Prediction
+<code style="color: #D85D43;">
+HEALTH /h:prediction 
+</code>
+
+The sequence diagram below illustrates the process of period prediction.
+
+1. The Handler class receives `userInput` through `Handler.processInput()` which calls `Handler.handleHealth` to handle health-related operations.
+2. `Parser.extractSubstringaFromSpecificIndex()` is then called to extract `typeOfHealth`, in this case, `prediction`. 
+3. The Handler class then calls `Parser.parsePredictionInput()`.
+4. `HealthList.getPeriodSize()` is then called to retrieve the size of `Periods` to ensure sufficient `Period` inputs are present for prediction.
+5. `HealthList.printLatestThreeCycles()` prints the user's latest three menstrual cycles.
+6. `HealthList.predictNextPeriodStartDate()` gets the predicted start date by calling `Period.nextCyclePrediction()`.
+7. `Period.getLastThreeCycleLengths()` is called to find the sum of the latest three cycle lengths before returning to `Period.nextCyclePrediction()` for necessary calculation to obtain the predicted start date.
+8. The `Parser` class calls `Period.printNextCyclePrediction()` with the predicted start date as a parameter. This method prints a string indicating the number of days until the predicted start date of the next period, or how many days late the period is if the current date is after the predicted start date.
+![Sequence Diagram for Add Period](img/period_prediction.png)
```

Comment:
![Screenshot 2024-04-04 124035](https://github.com/nus-cs2113-AY2324S2/tp/assets/110781598/f0534ba8-8470-4129-9aee-e7a9036a8d28)

Is it better to include the parenthesis and/or parameters behind the function name?